build/crds/router/router_v1_ruleEndpoint.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: ruleendpoints.rules.kubeedge.io
+spec:
+  group: rules.kubeedge.io
+  versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                ruleEndpointType:
```

Comment:
the word `ruleEndpoint` of `ruleEndpointType` seems unnecessary, since it is already in context ruleendpoint spec.

Reply:
TYPE is a key word in crd yaml, so use ruleendpointtypr here...

Reply:
you can define a field `type` for a object: 
``` yaml
spec:
  type: object
  properties:
      type:
         type: string
```

Reply:
@fisherxu what's your opinion about this?

Reply:
> @fisherxu what's your opinion about this?

We should use kubebuilder things to generate the crd yaml, it's hard to maintain manually.